cvmfs/util/algorithm.h

```diff
@@ -108,6 +115,59 @@ class StopWatch : SingleCopy {
 };
 
 
+class Log2Hist
+{
+ private:
+  uint number_of_bins;
+  atomic_int32 *bins;
```

Comment:
Use an std::vector<atomic_int32>.  This will make the `number_of_bins` field obsolete, too

Reply:
Done!